### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by Akbarable** [fkodom/dilated-attention-pytorch#1]
I want to use the LongNetLM and train it on some dummy questions and answer data.

Do I have to make changes in the encoder decoder layers in order to train this transformer? 

Thanks

**Comment by fkodom** [comment]
@Akbarable You shouldn't need to?  Gradients and everything work just fine with dilated attention. Compute cross-entropy loss, and you should be good to go. 😎 

^^ More than happy to discuss, if you have other questions.

**Comment by Akbarable** [comment]
Oh that's perfect then. I'm trying to train the model you created on text data, and my goal is to use it for inference and verify how many input tokens does the transformer accept on my GPU.

I would be thankful if you could share some outlines regarding how to go about training this on a dataset, I'm very new to transformers and I have trained a tranformer for language translation only once before, and am encountering a lot of errors when I'm trying to work with this.

Just some basic guidlines or suggestions would get me a long way. Thank you for creating this repo!



> @Akbarable You shouldn't need to?  Gradients and everything work just fine with dilated attention. Compute cross-entropy loss, and you should be good to go. 😎 
> 
> 
> 
> ^^ More than happy to discuss, if you have other questions.

**Comment by fkodom** [comment]
@Akbarable Sure thing -- an example training script is on my to-do list 😄 

Do you need LongNet for your use case?  Unless your text inputs are extremely long (more than a few thousand words), there are more established Transformer models that would work for you.  In that case, I would recommend looking into `huggingface/transformers`.  For example, here is a tutorial on fine-tuning a pretrained model from their library: https://huggingface.co/docs/transformers/training

**Comment by Akbarable** [comment]
Yup, I need to use LongNet specifically. My text input will consist of a considerable amount of tokens (definitely more than a few thousand). I'll keep an eye out for your training script then! 

I assume much of the steps would be similar to regular training. Thanks for sharing the HF link, I'll check it out and see if I can come up with anything. 😁

> @Akbarable Sure thing -- an example training script is on my to-do list 😄 
> 
> 
> 
> Do you need LongNet for your use case?  Unless your text inputs are extremely long (more than a few thousand words), there are more established Transformer models that would work for you.  In that case, I would recommend looking into `huggingface/transformers`.  For example, here is a tutorial on fine-tuning a pretrained model from their library: https://huggingface.co/docs/transformers/training
